starknet_py/contract.py

```diff
@@ -557,16 +556,16 @@ async def from_address(
         address = parse_address(address)
         proxy_config = Contract._create_proxy_config(proxy_config)
 
-        try:
-            abi = await ContractAbiResolver(
-                address=address, client=client, proxy_config=proxy_config
-            ).resolve()
-        except UnsupportedAbiError as err:
-            raise ValueError(
-                "Provided address of Cairo1 contract which is currently not supported in Contract."
-            ) from err
+        abi, cairo_version = await ContractAbiResolver(
```

Comment:
Would it be possible to not change the signature and logic, and instead write a static method in the Resolver that would return the cairo_version, given an abi?

Nothing would change then, except Contract constructor below would take 
```
ContractAbiResolver._cairo_version_from_abi(abi)
```